src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodTranslator.cs

```diff
@@ -51,18 +53,22 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
                 return null;
 
             var operand = arguments[0];
-            Type operandElementType;
 
-            if (operand.Type.IsArray)
-            {
-                operandElementType = operand.Type.GetElementType();
-            }
-            else if (operand.Type.IsGenericType && operand.Type.GetGenericTypeDefinition() == typeof(List<>))
+            var operandElementType = operand.Type.IsArray
+                ? operand.Type.GetElementType()
+                : operand.Type.IsGenericType && operand.Type.GetGenericTypeDefinition() == typeof(List<>)
+                    ? operand.Type.GetGenericArguments()[0]
+                    : null;
+
+            if (operandElementType == null)  // Not an array/list
+                return null;
+
+            // Even if the CLR type is an array/list, it may be mapped to a non-array database type (e.g. via value converters).
+            if (operand.TypeMapping is RelationalTypeMapping typeMapping &&
+                !(typeMapping is NpgsqlArrayTypeMapping) && !(typeMapping is NpgsqlJsonTypeMapping))
```

Comment:
Why are you checking for JSON mapping here?

Reply:
Because JSON documents also contain arrays which aren't mapped as NpgsqlArrayTypeMapping (JSON arrays and PG arrays aren't the same).